components/org.wso2.carbon.deployment.notifier/src/test/java/org/wso2/carbon/deployment/notifier/deployers/FaultyDeployer1.java

```diff
@@ -68,7 +78,11 @@ public void init() {
 
     public String deploy(Artifact artifact) throws CarbonDeploymentException {
         logger.info("Deploying : " + artifact.getName());
-        sample1Deployed = false;
+        if (artifact.getName().contains("sample1.txt")) {
```

Comment:
Possible NPE with calling contains on artifact.getName() ? Better checking for nullity before calling .contains